RAWDATA=raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L001_R1_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L001_R2_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L002_R1_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L002_R2_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L001_R1_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L001_R2_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L002_R1_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L002_R2_001.fastq.gz raw_data/Sample_C12_rep3/C12_rep3_ACAGTG_L001_R1_001.fastq.gz raw_data/Sample_C12_rep3/C12_rep3_ACAGTG_L001_R2_001.fastq.gz raw_data/Sample_C12_rep3/C12_rep3_ACAGTG_L002_R1_001.fastq.gz raw_data/Sample_C12_rep3/C12_rep3_ACAGTG_L002_R2_001.fastq.gz raw_data/Sample_C12_rep4/C12_rep4_GCCAAT_L001_R1_001.fastq.gz raw_data/Sample_C12_rep4/C12_rep4_GCCAAT_L001_R2_001.fastq.gz raw_data/Sample_C12_rep4/C12_rep4_GCCAAT_L002_R1_001.fastq.gz raw_data/Sample_C12_rep4/C12_rep4_GCCAAT_L002_R2_001.fastq.gz raw_data/Sample_C24_rep1/C24_rep1_ATCACG_L001_R1_001.fastq.gz raw_data/Sample_C24_rep1/C24_rep1_ATCACG_L001_R2_001.fastq.gz raw_data/Sample_C24_rep1/C24_rep1_ATCACG_L002_R1_001.fastq.gz raw_data/Sample_C24_rep1/C24_rep1_ATCACG_L002_R2_001.fastq.gz raw_data/Sample_C24_rep2/C24_rep2_TTAGGC_L001_R1_001.fastq.gz raw_data/Sample_C24_rep2/C24_rep2_TTAGGC_L001_R2_001.fastq.gz raw_data/Sample_C24_rep2/C24_rep2_TTAGGC_L002_R1_001.fastq.gz raw_data/Sample_C24_rep2/C24_rep2_TTAGGC_L002_R2_001.fastq.gz raw_data/Sample_C24_rep3/C24_rep3_ACTTGA_L001_R1_001.fastq.gz raw_data/Sample_C24_rep3/C24_rep3_ACTTGA_L001_R2_001.fastq.gz raw_data/Sample_C24_rep3/C24_rep3_ACTTGA_L002_R1_001.fastq.gz raw_data/Sample_C24_rep3/C24_rep3_ACTTGA_L002_R2_001.fastq.gz raw_data/Sample_C24_rep4/C24_rep4_GATCAG_L001_R1_001.fastq.gz raw_data/Sample_C24_rep4/C24_rep4_GATCAG_L001_R2_001.fastq.gz raw_data/Sample_C24_rep4/C24_rep4_GATCAG_L002_R1_001.fastq.gz raw_data/Sample_C24_rep4/C24_rep4_GATCAG_L002_R2_001.fastq.gz raw_data/Sample_E12_rep1/E12_rep1_CAGATC_L001_R1_001.fastq.gz raw_data/Sample_E12_rep1/E12_rep1_CAGATC_L001_R2_001.fastq.gz raw_data/Sample_E12_rep1/E12_rep1_CAGATC_L002_R1_001.fastq.gz raw_data/Sample_E12_rep1/E12_rep1_CAGATC_L002_R2_001.fastq.gz raw_data/Sample_E12_rep2/E12_rep2_CTTGTA_L001_R1_001.fastq.gz raw_data/Sample_E12_rep2/E12_rep2_CTTGTA_L001_R2_001.fastq.gz raw_data/Sample_E12_rep2/E12_rep2_CTTGTA_L002_R1_001.fastq.gz raw_data/Sample_E12_rep2/E12_rep2_CTTGTA_L002_R2_001.fastq.gz raw_data/Sample_E12_rep3/E12_rep3_AGTCAA_L001_R1_001.fastq.gz raw_data/Sample_E12_rep3/E12_rep3_AGTCAA_L001_R2_001.fastq.gz raw_data/Sample_E12_rep3/E12_rep3_AGTCAA_L002_R1_001.fastq.gz raw_data/Sample_E12_rep3/E12_rep3_AGTCAA_L002_R2_001.fastq.gz raw_data/Sample_E12_rep4/E12_rep4_AGTTCC_L001_R1_001.fastq.gz raw_data/Sample_E12_rep4/E12_rep4_AGTTCC_L001_R2_001.fastq.gz raw_data/Sample_E12_rep4/E12_rep4_AGTTCC_L002_R1_001.fastq.gz raw_data/Sample_E12_rep4/E12_rep4_AGTTCC_L002_R2_001.fastq.gz raw_data/Sample_E24_rep1/E24_rep1_TAGCTT_L001_R1_001.fastq.gz raw_data/Sample_E24_rep1/E24_rep1_TAGCTT_L001_R2_001.fastq.gz raw_data/Sample_E24_rep1/E24_rep1_TAGCTT_L002_R1_001.fastq.gz raw_data/Sample_E24_rep1/E24_rep1_TAGCTT_L002_R2_001.fastq.gz raw_data/Sample_E24_rep2/E24_rep2_GGCTAC_L001_R1_001.fastq.gz raw_data/Sample_E24_rep2/E24_rep2_GGCTAC_L001_R2_001.fastq.gz raw_data/Sample_E24_rep2/E24_rep2_GGCTAC_L002_R1_001.fastq.gz raw_data/Sample_E24_rep2/E24_rep2_GGCTAC_L002_R2_001.fastq.gz raw_data/Sample_E24_rep3/E24_rep3_GTGGCC_L001_R1_001.fastq.gz raw_data/Sample_E24_rep3/E24_rep3_GTGGCC_L001_R2_001.fastq.gz raw_data/Sample_E24_rep3/E24_rep3_GTGGCC_L002_R1_001.fastq.gz raw_data/Sample_E24_rep3/E24_rep3_GTGGCC_L002_R2_001.fastq.gz raw_data/Sample_E24_rep4/E24_rep4_GTTTCG_L001_R1_001.fastq.gz raw_data/Sample_E24_rep4/E24_rep4_GTTTCG_L001_R2_001.fastq.gz raw_data/Sample_E24_rep4/E24_rep4_GTTTCG_L002_R1_001.fastq.gz raw_data/Sample_E24_rep4/E24_rep4_GTTTCG_L002_R2_001.fastq.gz raw_data/Sample_T12_rep1/T12_rep1_ATGTCA_L001_R1_001.fastq.gz raw_data/Sample_T12_rep1/T12_rep1_ATGTCA_L001_R2_001.fastq.gz raw_data/Sample_T12_rep1/T12_rep1_ATGTCA_L002_R1_001.fastq.gz raw_data/Sample_T12_rep1/T12_rep1_ATGTCA_L002_R2_001.fastq.gz raw_data/Sample_T12_rep2/T12_rep2_CCGTCC_L001_R1_001.fastq.gz raw_data/Sample_T12_rep2/T12_rep2_CCGTCC_L001_R2_001.fastq.gz raw_data/Sample_T12_rep2/T12_rep2_CCGTCC_L002_R1_001.fastq.gz raw_data/Sample_T12_rep2/T12_rep2_CCGTCC_L002_R2_001.fastq.gz raw_data/Sample_T12_rep3/T12_rep3_GTCCGC_L001_R1_001.fastq.gz raw_data/Sample_T12_rep3/T12_rep3_GTCCGC_L001_R2_001.fastq.gz raw_data/Sample_T12_rep3/T12_rep3_GTCCGC_L002_R1_001.fastq.gz raw_data/Sample_T12_rep3/T12_rep3_GTCCGC_L002_R2_001.fastq.gz raw_data/Sample_T12_rep4/T12_rep4_GTGAAA_L001_R1_001.fastq.gz raw_data/Sample_T12_rep4/T12_rep4_GTGAAA_L001_R2_001.fastq.gz raw_data/Sample_T12_rep4/T12_rep4_GTGAAA_L002_R1_001.fastq.gz raw_data/Sample_T12_rep4/T12_rep4_GTGAAA_L002_R2_001.fastq.gz raw_data/Sample_T24_rep1/T24_rep1_CGTACG_L001_R1_001.fastq.gz raw_data/Sample_T24_rep1/T24_rep1_CGTACG_L001_R2_001.fastq.gz raw_data/Sample_T24_rep1/T24_rep1_CGTACG_L002_R1_001.fastq.gz raw_data/Sample_T24_rep1/T24_rep1_CGTACG_L002_R2_001.fastq.gz raw_data/Sample_T24_rep2/T24_rep2_GAGTGG_L001_R1_001.fastq.gz raw_data/Sample_T24_rep2/T24_rep2_GAGTGG_L001_R2_001.fastq.gz raw_data/Sample_T24_rep2/T24_rep2_GAGTGG_L002_R1_001.fastq.gz raw_data/Sample_T24_rep2/T24_rep2_GAGTGG_L002_R2_001.fastq.gz raw_data/Sample_T24_rep3/T24_rep3_ACTGAT_L001_R1_001.fastq.gz raw_data/Sample_T24_rep3/T24_rep3_ACTGAT_L001_R2_001.fastq.gz raw_data/Sample_T24_rep3/T24_rep3_ACTGAT_L002_R1_001.fastq.gz raw_data/Sample_T24_rep3/T24_rep3_ACTGAT_L002_R2_001.fastq.gz raw_data/Sample_T24_rep4/T24_rep4_ATTCCT_L001_R1_001.fastq.gz raw_data/Sample_T24_rep4/T24_rep4_ATTCCT_L001_R2_001.fastq.gz raw_data/Sample_T24_rep4/T24_rep4_ATTCCT_L002_R1_001.fastq.gz raw_data/Sample_T24_rep4/T24_rep4_ATTCCT_L002_R2_001.fastq.gz

#Small set of data files used for testing
#RAWDATA=raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L001_R1_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L001_R2_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L002_R1_001.fastq.gz raw_data/Sample_C12_rep1/C12_rep1_CGATGT_L002_R2_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L001_R1_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L001_R2_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L002_R1_001.fastq.gz raw_data/Sample_C12_rep2/C12_rep2_TGACCA_L002_R2_001.fastq.gz

#QC raw data - do this pre-concatenation to check for lane issues

QC_OUTDIR=fastqc_out
QC_RAW_OUTDIR=$(addsuffix /raw, $(QC_OUTDIR))

qc_out:
	mkdir $(QC_OUTDIR)

qc_raw_out: qc_out
	mkdir $(QC_RAW_OUTDIR)

qc_raw: $(RAWDATA) qc_raw_out
	fastqc -o $(QC_RAW_OUTDIR) -t $(words $(RAWDATA))  $(RAWDATA)

qc_clean: $(QC_OUTDIR)
	rm -r $(QC_OUTDIR)

#Concatenate raw data files over lanes
#
#Because some programs do not handle gzipped fastq, decompres at this point

DATADIRS=$(dir $(RAWDATA))
BASENAMES=$(sort $(subst /, , $(subst raw_data/Sample_, , $(DATADIRS))))
CONCAT_BASE=concatenated_raw/
CONCAT_DIRS=$(addprefix $(CONCAT_BASE), $(BASENAMES)) 
CONCAT_FILES=$(join $(addsuffix /, $(CONCAT_DIRS)), $(addsuffix _R1.fastq, $(BASENAMES)))
CONCAT_FILES+=$(join $(addsuffix /, $(CONCAT_DIRS)), $(addsuffix _R2.fastq, $(BASENAMES)))


$(CONCAT_FILES): $(RAWDATA)
	if [ ! -d $(CONCAT_BASE) ]; then mkdir $(CONCAT_BASE); fi
	for d in $(CONCAT_DIRS); do \
		if [ ! -d $$d ]; then mkdir $$d; fi; \
	done
	for b in $(BASENAMES); do \
		zcat raw_data/Sample_$$b/$$b*R1*.fastq.gz > concatenated_raw/$$b/$$b\_R1.fastq; \
		zcat raw_data/Sample_$$b/$$b*R2*.fastq.gz > concatenated_raw/$$b/$$b\_R2.fastq; \
	done

concat_raw: $(CONCAT_FILES)

concat_clean: $(CONCAT_BASE)
	rm -r $(CONCAT_BASE)

#Check that concatenated files are still correctly paired

PAIR_CHECK_FILE=$(addprefix $(CONCAT_BASE), pairingcheck.txt)

$(PAIR_CHECK_FILE): $(CONCAT_FILES)
	if [ -f $(PAIR_CHECK_FILE) ]; then rm $(PAIR_CHECK_FILE); fi; \
	for sample in $(BASENAMES); do \
		echo $$sample >> $(PAIR_CHECK_FILE); \
		pe-sync-2-files.pl $(CONCAT_BASE)/$$sample/*.fastq >> $(PAIR_CHECK_FILE); \
	done

concat_check: $(PAIR_CHECK_FILE)

#
# Use trimmomatic to remove adaptors and low quality regions
#

#Set up the input and output files. This gets a little messy because of the number of
#files. Would be nice to use trimmomatics -basein and -baseout options by the rules
#used to parse these are not well documented. 

R1IN=$(join $(addsuffix /, $(CONCAT_DIRS)), $(addsuffix _R1.fastq, $(BASENAMES)))
R2IN=$(join $(addsuffix /, $(CONCAT_DIRS)), $(addsuffix _R2.fastq, $(BASENAMES)))

TRIM_BASE=trimmed/
TRIM_DIRS=$(addprefix $(TRIM_BASE), $(BASENAMES)) 

#Paired output
1POUT=$(join $(addsuffix /, $(TRIM_DIRS)), $(addsuffix _R1_P.fastq, $(BASENAMES)))
2POUT=$(join $(addsuffix /, $(TRIM_DIRS)), $(addsuffix _R2_P.fastq, $(BASENAMES)))

#Unpaired output
1UOUT=$(join $(addsuffix /, $(TRIM_DIRS)), $(addsuffix _R1_U.fastq, $(BASENAMES)))
2UOUT=$(join $(addsuffix /, $(TRIM_DIRS)), $(addsuffix _R2_U.fastq, $(BASENAMES)))

TRIM_FILES=$(1POUT) $(2POUT) $(1UOUT) $(2UOUT)

TRIM_CMD=java -jar /usr/local/Trimmomatic-0.36/trimmomatic-0.36.jar PE

###################################
#                                 #
# Adjust trimmomatic options here #
#                                 #
###################################
TRIM_OPS=ILLUMINACLIP:/usr/local/Trimmomatic-0.36/adapters/TruSeq3-PE-2.fa:3:25:7 SLIDINGWINDOW:6:20 MINLEN:40

$(TRIM_FILES): $(CONCAT_FILES)
	if [ ! -d $(TRIM_BASE) ]; then mkdir $(TRIM_BASE); fi
	for d in $(TRIM_DIRS); do \
		if [ ! -d $$d ]; then mkdir $$d; fi; \
	done;
	$(foreach sample, $(BASENAMES), $(TRIM_CMD) $(filter concatenated_raw/$(sample)%, $(R1IN)) $(filter concatenated_raw/$(sample)%, $(R2IN)) $(filter trimmed/$(sample)%, $(1POUT)) $(filter trimmed/$(sample)%, $(1UOUT)) $(filter trimmed/$(sample)%, $(2POUT)) $(filter trimmed/$(sample)%, $(2UOUT)) $(TRIM_OPS);)


trim: $(TRIM_FILES)

trim_clean: $(TRIM_BASE)
	rm -r $(TRIM_BASE)


#
# Run QC on trimmed data
#

QC_TRIMMED_OUTDIR=$(addsuffix /trimmed, $(QC_OUTDIR))

qc_trimmed: $(TRIM_FILES)
	if [ ! -d $(QC_OUTDIR) ]; then mkdir $(QC_OUTDIR); fi
	if [ ! -d $(QC_TRIMMED_OUTDIR) ]; then mkdir $(QC_TRIMMED_OUTDIR); fi
	fastqc -o $(QC_TRIMMED_OUTDIR) -t $(words $(TRIM_FILES))  $(TRIM_FILES)

#
# map reads with GSNAP
#

#Set up the indexed genome database for GSNAP
GENOME_DIR=genome

GENOME_DB=SBAGenome

GENOME_DB_DIR=$(GENOME_DIR)/$(GENOME_DB)

GENOME_FASTA=$(GENOME_DIR)/SBAphid_ref_genome_v6.fna

GENOME_DB_FILES=$(addprefix $(GENOME_DB_DIR)/, SBAGenome.chromosome SBAGenome.chromosome.iit SBAGenome.chrsubset SBAGenome.contig SBAGenome.contig.iit SBAGenome.genomebits128 SBAGenome.genomecomp SBAGenome.ref153offsets64meta SBAGenome.ref153offsets64strm SBAGenome.ref153positions SBAGenome.sachildexc SBAGenome.sachildguide1024 SBAGenome.saindex64meta SBAGenome.saindex64strm SBAGenome.salcpchilddc SBAGenome.salcpexc SBAGenome.salcpguide1024 SBAGenome.sarray SBAGenome.version SBAGenome.maps)


$(GENOME_DB_FILES): $(GENOME_FASTA)
	gmap_build -D $(GENOME_DIR) -d $(GENOME_DB) $(GENOME_FASTA)

genome_db: $(GENOME_DB_FILES)

genome_db_clean: $(GENOME_DB_DIR)
	rm -r $(GENOME_DB_DIR)

#Align paired reads and write SAM file

SAMPLES=$(BASENAMES)

ALIGN_DIR=bam

GSNAP_CMD=gsnap -t 26 -N 1 -A sam

BAM_FILES=$(addsuffix .bam, $(addprefix $(ALIGN_DIR)/, $(SAMPLES)))

$(BAM_FILES): $(1POUT) $(2POUT)
	if [ ! -d $(ALIGN_DIR) ]; then mkdir $(ALIGN_DIR); fi
	$(foreach sample, $(SAMPLES), $(GSNAP_CMD) -D $(GENOME_DIR) -d $(GENOME_DB) $(TRIM_BASE)$(sample)/$(sample)_R1_P.fastq $(TRIM_BASE)$(sample)/$(sample)_R2_P.fastq | samtools view -bh > $(ALIGN_DIR)/$(sample).bam;)

align: $(BAM_FILES)

align_clean: $(ALIGN_DIR)
	rm -r $(ALIGN_DIR)

#Sort and index bam files
BAM_FILES_SORTED=$(subst .bam,.sorted.bam, $(BAM_FILES))

$(BAM_FILES_SORTED): $(BAM_FILES)
	$(foreach bamfile, $(BAM_FILES), samtools sort -@ 28 $(bamfile) > $(subst .bam,.sorted.bam, $(bamfile));)

align_sorted: $(BAM_FILES_SORTED)

BAM_INDEX_FILES=$(addsuffix .bai, $(BAM_FILES_SORTED))

$(BAM_INDEX_FILES): $(BAM_FILES_SORTED)
	$(foreach bamfile, $(BAM_FILES_SORTED), samtools index -@ 28 $(bamfile);)

align_index: $(BAM_INDEX_FILES)

#
# Use stringtie to assemble transcripts that are not in the genome annotation
#
STRINGTIE_DIR=stringtie

STRINGTIE_REF=genome/SBAphid_ref_genome_v2.gff3

STRINGTIE_CMD=stringtie -p 26 -G $(STRINGTIE_REF)

STRINGTIE_FILES=$(addsuffix .gtf, $(addprefix $(STRINGTIE_DIR)/, $(SAMPLES)))

$(STRINGTIE_FILES): $(BAM_FILES_SORTED)
	if [ ! -d $(STRINGTIE_DIR) ]; then mkdir $(STRINGTIE_DIR); fi
	$(foreach bamfile, $(BAM_FILES_SORTED), $(STRINGTIE_CMD) -o $(subst $(ALIGN_DIR)/, $(STRINGTIE_DIR)/, $(subst .sorted.bam,.gtf, $(bamfile))) $(bamfile);)


assemble_transcripts: $(STRINGTIE_FILES)

STRINGTIE_MERGED=$(STRINGTIE_DIR)/stringtie_merged.gtf

$(STRINGTIE_MERGED): $(STRINGTIE_FILES)
	stringtie --merge -G $(STRINGTIE_REF) -o $(STRINGTIE_MERGED) $(STRINGTIE_FILES)

merge_transcripts: $(STRINGTIE_MERGED)

#
#Compare merged assembly to original annotation 
#

GFF_CMP_FILES=$(addprefix $(STRINGTIE_DIR)/, gffcmp.annotated.gtf  gffcmp.loci  gffcmp.stats  gffcmp.tracking)

$(GFF_CMP_FILES): $(STRINGTIE_MERGED) $(STRINGTIE_REF)
	gffcompare -r $(STRINGTIE_REF) $(STRINGTIE_MERGED)
	mv gffcmp.* $(STRINGTIE_DIR)

analyze_merged: $(GFF_CMP_FILES)

#
#Convert orginal annotation to GTF
#

ANNOT_DIR=annotations

REF_GTF= $(addprefix $(ANNOT_DIR)/, SBAphid_ref_genome_v2.gtf)

$(REF_GTF): $(STRINGTIE_REF)
	if [ ! -d $(ANNOT_DIR) ]; then mkdir $(ANNOT_DIR); fi
	gffread -T -o $(REF_GTF) $(STRINGTIE_REF)

STRINGTIE_UNIQUE= $(addprefix $(ANNOT_DIR)/, stringtie_unique.gtf)

$(STRINGTIE_UNIQUE): $(STRINGTIE_MERGED) $(REF_GTF)
	bedtools intersect -v -b $(REF_GTF) -a $(STRINGTIE_MERGED) > $(STRINGTIE_UNIQUE)

UNIQUE_ANNOTATIONS= $(addprefix $(ANNOT_DIR)/, unique.gtf)

$(UNIQUE_ANNOTATIONS): $(REF_GTF) $(STRINGTIE_UNIQUE)
	cat $(REF_GTF) $(STRINGTIE_UNIQUE) > $(UNIQUE_ANNOTATIONS)

unique_annots: $(UNIQUE_ANNOTATIONS)

annots_clean: $(ANNOT_DIR)
	rm -r $(ANNOT_DIR)



#
#Generate fragment counts per gene for differential gene expression anlysis with DESeq2
#

FRAG_COUNTS_DIR= fragmentcounts

FRAG_COUNTS_TABLE= $(addprefix $(FRAG_COUNTS_DIR)/, count_table)

$(FRAG_COUNTS_TABLE): $(UNIQUE_ANNOTATIONS) $(BAM_FILES_SORTED)
	if [ ! -d $(FRAG_COUNTS_DIR) ]; then mkdir $(FRAG_COUNTS_DIR); fi
	featureCounts -p -T 26 -a $(UNIQUE_ANNOTATIONS) -t exon -g gene_id -o $(FRAG_COUNTS_TABLE) $(BAM_FILES_SORTED)

frag_counts: $(FRAG_COUNTS_TABLE)

frag_counts_clean: $(FRAG_COUNTS_DIR)
	rm -r $(FRAG_COUNTS_DIR)
#
#Use Stringtie to count reads mapping to exons/transcripts/genes
#For use with ballgown. Doesn't give us counts for DESeq, but has some nice visualization tools etc.
#

BALLGOWN_DIR=ballgown

BALLGOWN_SUB_DIRS=$(addprefix $(BALLGOWN_DIR)/, $(SAMPLES))

$(BALLGOWN_SUB_DIRS): $(STRINGTIE_MERGED) $(BAM_FILES_SORTED)
	if [ ! -d $(BALLGOWN_DIR) ]; then mkdir $(BALLGOWN_DIR); fi
	$(foreach sample, $(SAMPLES), stringtie -p 26 -e -b $(addprefix $(BALLGOWN_DIR)/, $(sample)) -G $(STRINGTIE_MERGED) $(addsuffix .sorted.bam, $(addprefix $(ALIGN_DIR)/, $(sample))); )

ballgown: $(BALLGOWN_SUB_DIRS)

ballgown_clean: $(BALLGOWN_DIR)
	rm -r $(BALLGOWN_DIR)
